src/diskio/threaded.rs

```diff
@@ -115,16 +115,18 @@ impl<'a> Executor for Threaded<'a> {
         // items, and the download tracker's progress is confounded with
         // actual handling of data today, we synthesis a data buffer and
         // pretend to have bytes to deliver.
-        self.notify_handler
-            .map(|handler| handler(Notification::DownloadFinished));
-        self.notify_handler
-            .map(|handler| handler(Notification::DownloadPushUnits("iops")));
+        if let Some(handler) = self.notify_handler {
+            handler(Notification::DownloadFinished);
+        }
+        if let Some(handler) = self.notify_handler {
+            handler(Notification::DownloadPushUnits("iops"));
+        }
         let mut prev_files = self.n_files.load(Ordering::Relaxed);
-        self.notify_handler.map(|handler| {
+        if let Some(handler) = self.notify_handler {
             handler(Notification::DownloadContentLengthReceived(
                 prev_files as u64,
-            ))
-        });
+            ));
+        }
```

Comment:
In fact, move the `let mut prev_files...` above the `if let` above and it can be one `if` with three calls in it perhaps?